Makefile

```diff
@@ -30,96 +33,88 @@ INSTALL_CONSUMER_CONFIG_DIR := ${INSTALL_CONFIG_HOME_DIR}/consumer
 
-build-protobuf:
-ifeq ($(PROTOC),)
-	make rebuild-protobuf
-else
-	@echo 'protoc is already built. Run make rebuild-protobuf to remove existing one and rebuild it.'
+$(PROTOC):
+	mkdir -p $(PROTOC_DIR)
+ifdef mac-os-host
+	wget -O $(PROTOC_DIR)/protoc.zip https://github.com/protocolbuffers/protobuf/releases/download/v3.19.4/protoc-3.19.4-osx-x86_64.zip
 endif
+ifdef linux-os-host
+	wget -O $(PROTOC_DIR)/protoc.zip https://github.com/protocolbuffers/protobuf/releases/download/v3.19.4/protoc-3.19.4-linux-x86_64.zip
+endif
+	cd $(PROTOC_DIR) && unzip protoc.zip
+	chmod +x $(PROTOC)
 
 $(PROTOC_GEN_GO):
 	go install github.com/golang/protobuf/protoc-gen-go
 
-PROTOFILE_DIR := $(abspath proto)
+PROTO_DIR := ./proto
+PROTO_TARGETS := $(PROTO_DIR)/api.pb.go $(PROTO_DIR)/data.pb.go
+.SUFFIXES: .proto .pb.go
+%.pb.go: %.proto $(PROTOC_GEN_GO) $(PROTOC)
+	$(PROTOC) --proto_path=$(PROTO_DIR) --go_out=$(PROTO_DIR) $<
 
-compile-protobuf: $(PROTOC_GEN_GO) 
-	rm -f $(PROTOFILE_DIR)/*.go
-	protoc --proto_path=$(PROTOFILE_DIR) --go_out=$(PROTOFILE_DIR) $(PROTOFILE_DIR)/*.proto
+.PHONY: compile-protobuf prepare
+compile-protobuf: $(PROTO_TARGETS) $(PROTOC_GEN_GO) $(PROTOC)
 
-.PHONY: rebuild-rocksdb build-rocksdb
-rebuild-rocksdb:
-	rm -rf $(ROCKSDB_BUILD_DIR)
+$(LIB_ROCKSDB):
+	if [ -d $(GIT_DIR) ]; then \
+		git submodule update --init --recursive; \
+	fi
 	mkdir -p $(ROCKSDB_BUILD_DIR)
 ifdef mac-os-host
-	cd $(ROCKSDB_BUILD_DIR) && cmake .. $(BASE_CMAKE_FLAGS) -DCMAKE_BUILD_TYPE=RelWithDebInfo -DPORTABLE=ON -DWITH_TESTS=OFF \
-	-DWITH_BENCHMARK_TOOLS=OFF -DWITH_SNAPPY=ON -DUSE_RTTI=ON -DWITH_GFLAGS=OFF \
-	&& make -j $(NPROC) install
+	cd $(ROCKSDB_BUILD_DIR) &&\
+ 	cmake .. -DCMAKE_TARGET_MESSAGES=OFF -DCMAKE_BUILD_TYPE=RelWithDebInfo -DPORTABLE=ON -DWITH_TESTS=OFF \
+		-DWITH_BENCHMARK_TOOLS=OFF -DWITH_SNAPPY=ON -DUSE_RTTI=ON -DWITH_GFLAGS=OFF -DCMAKE_INSTALL_PREFIX=$(ROCKSDB_INSTALL_DIR) &&\
+	make -j$(NPROC)
 endif
 ifdef linux-os-host
-	cd $(ROCKSDB_BUILD_DIR) && cmake .. $(BASE_CMAKE_FLAGS) -DCMAKE_POSITION_INDEPENDENT_CODE=ON -DCMAKE_BUILD_TYPE=RelWithDebInfo \
-	-DPORTABLE=ON -DWITH_FALLOCATE=OFF -DWITH_TESTS=OFF -DWITH_BENCHMARK_TOOLS=OFF -DWITH_SNAPPY=ON -DUSE_RTTI=ON -DWITH_GFLAGS=OFF \
-	&& make -j $(NPROC) install
+	cd $(ROCKSDB_BUILD_DIR) &&\
+	cmake .. -DCMAKE_TARGET_MESSAGES=OFF -DCMAKE_POSITION_INDEPENDENT_CODE=ON -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+		-DPORTABLE=ON -DWITH_FALLOCATE=OFF -DWITH_TESTS=OFF -DWITH_BENCHMARK_TOOLS=OFF -DWITH_SNAPPY=ON -DUSE_RTTI=ON -DWITH_GFLAGS=OFF -DCMAKE_INSTALL_PREFIX=$(ROCKSDB_INSTALL_DIR) &&\
+	make -j$(NPROC)
 endif
+	cd $(ROCKSDB_BUILD_DIR) && make install
+	touch $(LIB_ROCKSDB)
 
-build-rocksdb:
-ifeq ("$(wildcard $(LIB_ROCKSDB))", "")
-	make rebuild-rocksdb
-else
-	@echo RocksDB is already built. Run make rebuild-rocksdb to remove existing one and rebuild it.
-endif
+.PHONY: build-rocksdb
+build-rocksdb: $(LIB_ROCKSDB)
 
-.PHONY: build-broker build-client
-build-broker:
+$(BROKER_BIN): $(LIB_ROCKSDB)
```

Comment:
Is this expression working well? 
I encountered an error like below when I typed `make build-broker`.
`make: Nothing to be done for `build-broker'.`

Reply:
After executing `make clean`, it works well